File: backend/app/schemas/credential.py
from pydantic import BaseModel, Field
from typing import Dict, Any, Optional
from datetime import datetime

class CredentialData(BaseModel):
    # This represents the payload of the credential for hashing
    student_reference: str
    examination_id: str
    credential_type: str
    institution_name: str
    grade: Optional[str] = None
    issue_date: str

class CredentialCreate(BaseModel):
    credential_id: str
    credential_data: CredentialData

class CredentialResponse(BaseModel):
    credential_id: str
    document_hash: str
    blockchain_tx_hash: Optional[str] = None
    status: str
    issued_at: datetime
    credential_data: Optional[CredentialData] = None

class CredentialVerificationRequest(BaseModel):
    credential_id: str
    credential_data: CredentialData

class CredentialVerificationResponse(BaseModel):
    credential_id: str
    result: str # VALID, TAMPERED, REVOKED, NOT_FOUND
    hash_match: bool
    blockchain_verified: bool
    revoked: bool

class CredentialRevokeRequest(BaseModel):
    reason: Optional[str] = None

File: frontend/src/types/credential.ts
export interface CredentialData {
  student_reference: string;
  examination_id: string;
  credential_type: string;
  institution_name: string;
  grade?: string;
  issue_date: string;
}

export interface CredentialCreate {
  credential_id: string;
  credential_data: CredentialData;
}

export interface CredentialResponse {
  credential_id: string;
  document_hash: string;
  blockchain_tx_hash?: string;
  status: string;
  issued_at: string;
  credential_data?: CredentialData;
}

export interface CredentialVerificationRequest {
  credential_id: string;
  credential_data: CredentialData;
}

export interface CredentialVerificationResponse {
  credential_id: string;
  result: 'VALID' | 'TAMPERED' | 'REVOKED' | 'NOT_FOUND';
  hash_match: boolean;
  blockchain_verified: boolean;
  revoked: boolean;
}
